README.md

```diff
@@ -31,37 +36,255 @@ npm install @uploadcare/upload-client --save
 
 ## Usage
 
+### High Level API
+
+To get access to High Level API you need to create 
+an instance of `UploadClient` and provide settings to him:
+
 ```javascript
 import UploadClient from '@uploadcare/upload-client'
 import {FileFromEnum} from '@uploadcare/upload-client'
 
 const client = new UploadClient({publicKey: 'YOUR_PUBLIC_KEY'})
+```
 
-client.api.request({path: 'info', query})
-  .then(response => console.log(response.data))
+After that you can upload file from binary data:
 
-client.api.info(uuid)
-  .then(data => console.log(data.is_image))
+```javascript
+const fileUpload = client.fileFrom(FileFromEnum.Object, fileData)
 
-const directUpload = client.api.base(fileData)
+fileUpload
+  .then(file => console.log(file.uuid))
+```
 
-directUpload
-  .then(data => console.log(data.file))
+Or from URL with file:
 
-directUpload.onProgress = (progressEvent) => console.log(progressEvent.loaded / progressEvent.total)
+```javascript
+const fileURL = 'https://example.com/file.jpg'
+const fileUpload = client.fileFrom(FileFromEnum.URL, fileURL)
 
-const fileUpload = client.fileFrom(FileFromEnum.Object, fileData)
+fileUpload
+  .then(file => console.log(file.uuid))
+```
+
+Or even from `uploaded` before to Uploadcare file:
+```javascript
+const fileUUID = 'edfdf045-34c0-4087-bbdd-e3834921f890'
+const fileUpload = client.fileFrom(FileFromEnum.Uploaded, fileUUID)
 
 fileUpload
   .then(file => console.log(file.uuid))
+```
 
+You can track uploading progress:
+```javascript
 fileUpload.onProgress = (progress => {
   console.log(progress.state)
   console.log(progress.uploaded.loaded / progress.uploaded.total)
   console.log(progress.value)
 })
 ```
 
+Or set callback function that will be called when file was uploaded:
+```javascript
+fileUpload.onUploaded = (uuid => console.log(`File "${uuid}" was uploaded.`))
+```
+
+Or when file is ready on CDN:
+```javascript
+fileUpload.onReady = (file => console.log(`File "${file.uuid}" is ready on CDN.`))
+```
+
+You can cancel file uploading and track this event:
```

Comment:
This also deserves a separate section: to provide extra context, i.e., when the cancelation is applicable: if a file `is_ready`, you'd need to use REST API calls to remove it rather than canceling the upload.

Reply:
Сancelation is applicable anytime you want